src/adapters/dweb-staking.adapter.ts

```diff
@@ -0,0 +1,71 @@
+import {BindingScope, extensionFor, injectable} from '@loopback/core';
+import {BigNumber} from 'ethers';
+import {STAKING_ADAPTERS_EXTENSION_POINT} from '../keys.js';
+import {BaseStakingContractAdapter, StakingAsset} from '../staking.js';
+// Use the full path to import instead of `../types`
+import {DwebStaking__factory} from '../types/factories/DwebStaking__factory.js';
+
+abstract class BaseDwebStakingContractAdapter extends BaseStakingContractAdapter {
+  /**
+   * Get staked token balance for the given owner
+   * @param owner - Owner address
+   * @returns
+   */
+  async getStakedTokenBalance(owner: string): Promise<BigNumber> {
+    const contract = DwebStaking__factory.connect(
+      this.contractAddress,
+      this.provider,
+    );
+    try {
+      return (await contract.getStakeDetails(owner)).initialDeposit;
+    } catch (err) {
+      return BigNumber.from(0);
```

Comment:
We should let the error throw instead of returning `0`. Otherwise, if the RPC provider fails, corresponding roles may be removed by mistake.